xarray/convert.py

```diff
@@ -181,7 +183,9 @@ def from_iris(cube):
     cell_methods = _iris_cell_methods_to_str(cube.cell_methods)
     if cell_methods:
         array_attrs['cell_methods'] = cell_methods
-    dataarray = DataArray(cube.data, coords=coords, name=name,
+
+    cube_data = ma.filled(cube.core_data(), get_fill_value(cube.dtype)) if hasattr(cube, 'core_data') else cube.data
```

Comment:
Likewise, this should not coerce all data into dask arrays.

Reply:
OK, I hadn't appreciated that dask wasn't a hard requirement. I could leave them as whatever type they're currently stored in - how would xarray cope with a dask masked array?